test/integration/test_integration__io.py

```diff
@@ -1107,3 +1107,18 @@ def test_non_rectilinear__skip_parse_coordinates(open_rasterio):
     assert xds.rio.shape == (10, 10)
     with rasterio.open(test_file) as rds:
         assert rds.transform == xds.rio.transform()
+
+
+def test_complex_dtype(tmp_path):
+    test_file = os.path.join(TEST_INPUT_DATA_DIR, "cint16.tif")
+    xds = rioxarray.open_rasterio(test_file)
+    assert xds.rio.shape == (100, 100)
+    assert xds.dtype == "complex64"
+
+    # NOTE: waiting on https://github.com/mapbox/rasterio/issues/2206
+    # tmp_output = tmp_path / "tmp_cint16.tif"
+    # with pytest.warns(SerializationWarning):
+    #    xds.rio.to_raster(str(tmp_output), dtype='complex_int16')
```

Comment:
I am thinking it would be good to introduce something like `_rasterio_dtype` in `encoding` and then check if the dtype is complex in `rio.to_raster` (just in case they change their dtype) and pass the value in the encoding to `rasterio` if it is.

Reply:
i attempted to do this, let me know if the revisions are what you had in mind